documentapi/src/test/java/com/yahoo/documentapi/messagebus/protocol/test/TestFileUtil.java

```diff
@@ -4,16 +4,27 @@
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.charset.Charset;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+
 public class TestFileUtil {
     protected static final String DATA_PATH = "./test/crosslanguagefiles";
 
     public static void writeToFile(String path, byte[] data) throws IOException {
-        try (FileOutputStream stream = new FileOutputStream(path)) {
+        // Write to a temporary file to avoid racing with cross-language tests reading the
+        // exact same file we're trying to write.
+        String tmpPath = path + ".tmp";
+        try (FileOutputStream stream = new FileOutputStream(tmpPath)) {
             stream.write(data);
+            stream.flush();
```

Comment:
This doesn't do anything for `FileOutputStream` and for buffered writers `flush()` should be called by `close()` anyway.

Reply:
Removed